source/steamshim/src/parent/parent.cpp

```diff
@@ -295,7 +295,10 @@ char *argv[MAX_NUM_ARGVS];
 static void ParseCommandLine( LPSTR lpCmdLine )
 {
 	argc = 1;
-	argv[0] = "exe";
+
+	static char buffer[MAX_PATH]={0};
+	GetModuleFileNameA(NULL,&argv[0],sizeof(buffer)/sizeof(*buffer));
+	argv[0] = buffer;
```

Comment:
Do you want the full path or only the filename?

And what is the expected result of the `if (strstr(*GArgv,"warfork_steam"))` stuff?

Reply:
I only *need* filename but it shouldn't cause an issue to use full path for it

the expected result of strstr is to determine whether it was launched through warfork_steam.exe, wf_server_steam.exe, or wftv_server_steam.exe

Reply:
Considering this source file is only ever added to one project (wf_steam) and the x64 version of that is `wf_steam_x64.exe` you will never find `warfork_steam` or `wftv_server` in the filename.

Reply:
the cmakelists will create symlinks to warfork_steam, wf_server_steam, wftv_server_steam with respective suffixes

Reply:
what should I do instead of links then?

Reply:
There are multiple options:
* Add a 'target' commandline option for wf_steam, that you can add to a link or batch file
* Create a small 'stub' application instead of the symlinks that launches wf_steam with the correct options
* Reverse the parent/child thing
   * (So warfork_x64 is the process being launched, and that launches wf_steam, that handles the communication with steam)

Reply:
A stub launcher is what I've seen done in most games that release on Steam PLUS another PC marketplace